hibernate-core/src/test/java/org/hibernate/orm/test/sql/ast/TupleInListAlternativeTests.java

```diff
@@ -0,0 +1,149 @@
+package org.hibernate.orm.test.sql.ast;
+
+import jakarta.persistence.*;
+import org.hibernate.dialect.DB2Dialect;
+
+import org.hibernate.testing.orm.junit.*;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItems;
+
+@DomainModel(annotatedClasses = {
+        TupleInListAlternativeTests.EntityWithEmbeddedSubEntity.class
+})
+@ServiceRegistry
+@SessionFactory
+@RequiresDialect(DB2Dialect.class)
```

Comment:
This test should be valid for all Dialects, please remove this `@RequiresDialect` annotation.